seatunnel-connectors-v2/connector-milvus/src/main/java/org/apache/seatunnel/connectors/seatunnel/milvus/sink/MilvusSinkWriter.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.seatunnel.connectors.seatunnel.milvus.state.MilvusCommitInfo;
+import org.apache.seatunnel.connectors.seatunnel.milvus.state.MilvusSinkState;
+
+import io.milvus.v2.client.ConnectConfig;
+import io.milvus.v2.client.MilvusClientV2;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import static org.apache.seatunnel.connectors.seatunnel.milvus.config.MilvusSinkConfig.BATCH_SIZE;
+
+@Slf4j
+/** MilvusSinkWriter is a sink writer that will write {@link SeaTunnelRow} to Milvus. */
+public class MilvusSinkWriter
+        implements SinkWriter<SeaTunnelRow, MilvusCommitInfo, MilvusSinkState> {
+    private final Context context;
+
+    private final ReadonlyConfig config;
+    private MilvusBatchWriter batchWriter;
+
+    public MilvusSinkWriter(
+            Context context,
+            CatalogTable catalogTable,
+            ReadonlyConfig config,
+            List<MilvusSinkState> milvusSinkStates) {
+        this.context = context;
+        this.config = config;
+        ConnectConfig connectConfig =
+                ConnectConfig.builder()
+                        .uri(config.get(MilvusSinkConfig.URL))
+                        .token(config.get(MilvusSinkConfig.TOKEN))
+                        .build();
+        this.batchWriter =
+                new MilvusBufferBatchWriter(
+                        catalogTable,
+                        config.get(BATCH_SIZE),
+                        getAutoId(catalogTable.getTableSchema().getPrimaryKey()),
+                        config.get(MilvusSinkConfig.ENABLE_UPSERT),
+                        new MilvusClientV2(connectConfig));
+    }
+
+    /**
+     * write data to third party data receiver.
+     *
+     * @param element the data need be written.
+     * @throws IOException throw IOException when write data failed.
+     */
+    @Override
+    public void write(SeaTunnelRow element) {
+        batchWriter.addToBatch(element);
+        if (batchWriter.needFlush()) {
+            batchWriter.flush();
+        }
+    }
+
+    private Boolean getAutoId(PrimaryKey primaryKey) {
+        if (null != primaryKey && null != primaryKey.getEnableAutoId()) {
+            return primaryKey.getEnableAutoId();
+        } else {
+            return config.get(MilvusSinkConfig.ENABLE_AUTO_ID);
+        }
+    }
+
+    /**
+     * prepare the commit, will be called before {@link #snapshotState(long checkpointId)}. If you
+     * need to use 2pc, you can return the commit info in this method, and receive the commit info
+     * in {@link SinkCommitter#commit(List)}. If this method failed (by throw exception), **Only**
+     * Spark engine will call {@link #abortPrepare()}
+     *
+     * @return the commit info need to commit
+     */
+    @Override
+    public Optional<MilvusCommitInfo> prepareCommit() throws IOException {
+        return Optional.empty();
```

Comment:
call `batchWriter.flush();`?


reference
https://github.com/apache/seatunnel/blob/dbdbdf015bceb90767787be465399c00b7baa167/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/sink/JdbcSinkWriter.java#L133

Reply:
Don't know it work, keep it current. Can you provider an interface to support batchWrite.